### ISSUE TITLE
DatabaseError on report save

### ISSUE THREAD

**Issue by unusual-thoughts** [openpolis/django-uwsgi-taskmanager#25]
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/uwsgidecorators.py", line 64, in manage_spool_request
    ret = f(*vars['args'], **vars['kwargs'])
  File "/usr/local/lib/python3.9/site-packages/taskmanager/tasks.py", line 107, in exec_command_task
    report_obj.save(
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 753, in save
    self.save_base(using=using, force_insert=force_insert,
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 790, in save_base
    updated = self._save_table(
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 877, in _save_table
    raise DatabaseError("Save with update_fields did not affect any rows.")
django.db.utils.DatabaseError: Save with update_fields did not affect any rows.
```

**Comment by unusual-thoughts** [comment]
somehow the original report has been deleted in the meantime, and this results in a new report object with an empty log save for the Started/Finished lines

**Comment by guglielmo** [comment]
I need a bit of information to try and help you.

How are you using the taskmanager (are you trying the demo project or are you using it inside one of your projects)?
Which django, and python releases are you using? I can see python 3.9, but what release of django?
When did this happen? While saving a report, but what kind of task where you running?

**Comment by unusual-thoughts** [comment]
Hi! I am using the taskmanager inside of my project
The issue doesn't happen with scheduled, recurring tasks.
The problem task is started this way (inside a view)
```python
Task.objects.filter(category__name="x", command__app_name="x", command__name="x", arguments="x").first().launch()
```
I am running django 3.1.4

**Comment by unusual-thoughts** [comment]
This is almost definitely due to `self.keep_last_n_reports()` being called right after spooling, in the original thread. This has a race condition that can delete the report that has just been created in the spooler process. So when the spooler process then tries to update the report, it is already gone.
So:
- `keep_last_n_reports` should probably be run just before spooling, or at the end of `exec_command_task`
- use a transaction and `select_for_update`, or remove `update_fields` in `report_obj.save`

**Comment by unusual-thoughts** [comment]
actually `keep_last_n_reports` is already run at the end of `exec_command_task`, so no need to run before spooling

**Comment by guglielmo** [comment]
Seems like a good one. Will merge.

**Comment by guglielmo** [comment]
Problem has beel fixed in release 2.2.9
